eng/containers/ci.yml

```diff
@@ -0,0 +1,39 @@
+pr: none
+
+trigger:
+  branches:
+    include:
+    - master
+    - main
+  paths:
+    include:
+    - eng/containers/
+    - eng/tools/test-proxy/dockerfile
+
+pool:
+  vmImage: 'ubuntu-18.04'
+
+variables:
+  containerRegistry: 'azsdkengsys'
+  imageTag: $(build.buildid)
+  DockerDeployments:
+  - name: test-proxy
+    dockerRepo: engsys/ubuntu_testproxy_server
+    dockerFile: ../../tools/test-proxy/dockerfile
+
+steps:
+- ${{ each config in variables.DockerDeployments }}:
+  - task: Docker@2
+    displayName: Build ${{ config.name }} Docker Image
+    inputs:
+      command: build
+      Dockerfile: ${{ config.dockerFile }}
+      tags: $(imageTag)
+      arguments: '-t $(containerRegistry).azurecr.io/${{ config.dockerRepo }}:$(imageTag)'
```

Comment:
Won't this require whoever wants to use the test proxy to have access to the ACR instance? How will this work in an OSS context?

Reply:
This is a very good point. Thanks for bringing it up. This is a good place to start though  :)